doc/Data Plane Code Generation/AzureSDKCodeGeneration_DataPlane_Quickstart.md

```diff
@@ -0,0 +1,195 @@
+
+We will use the [Azure SDK Generated Client template](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template-dpg) project as a reference. We have a template [swagger file](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template-dpg/Azure.Template.Generated/src/swagger) containing a REST API definition based on [Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). We will use the swagger to generate SDK using [autorest.csharp](https://github.com/Azure/autorest.csharp).
+
+**Learn more**: to understand more about Azure SDK Code Geneneration, see the [Azure SDK Code Geneneration docs](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/ProtocolMethods.md).
+
+This tutorial has following sections:
+
+- [Prerequisites](#prerequisites)
+- [Setup your repo](#setup-your-repo)
+- [Getting started](#getting-started)
+  - [Create the project](#create-the-project)
+  - [Create the generated layer](#create-the-generated-layer)
+- [Tests](#tests)
+- [Samples](#samples)
+- [Snippets](#snippets)
+- [Export Public API](#export-public-api)
+- [README](#readme)
+- [Changelog](#changelog)
+
+<!-- /TOC -->
+
+## Prerequisites
+
+- Install VS 2019 (Community or higher) and make sure you have the [latest updates](https://www.visualstudio.com/).
+  - Need at least .NET Framework 4.6.1 and 4.7 development tools
+- Install the **.NET Core cross-platform development** workloads in VisualStudio
+- Install **.NET Core 5.0.301 SDK** for your specific platform. (or a higher version within the 5.0.*** band)  (https://dotnet.microsoft.com/download/dotnet-core/5.0)
+- Install the latest version of [git](https://git-scm.com/downloads)
+- Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell), version 6 or higher.
+- Install [NodeJS](https://nodejs.org/) (14.x.x).
+
+## Setup your repo
+
+- Fork and clone an [azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net) repo. Follow the instructions in the [.NET Contributing.md](https://github.com/Azure/azure-sdk-for-net/issues/12903) to fork and clone the `azure-sdk-for-net` repo.
+- Create a branch to work in. 
+
+## Getting started
+
+For this tutorial, we'll create a getting started project in a branch of your fork of `azure-sdk-for-net` repo. To create the project, we'll copy the [Azure.Template.Generated](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template-dpg) project for .NET.
+
+The `Azure.Template.Generated` project is a great place to get started with Azure SDK .NET library development in general, as it contains a number of common patterns we use for tests, samples, documentation, generating API listings. It also has generated code demonstrating an example Data Plane Generated Client.
```

Comment:
In the template, src/properties/AssemblyInfo.cs, the NameSpace is not correct, shall we tell customer to correct it in this doc? And there are some Suppressions needed to added otherwise, build will fail. Shall we add those also?

Reply:
>In the src folder of template, there are a folder 'swagger' to store the swagger. Swagger should be in the azure-rest-api-specs, it should not be in the azure-sdk-for-net folder. It may be good not to add Swagger in the template

This is just a template project, in the template swagger I added all the CRUD operations which follows the REST API guidelines which gives the partner teams idea about the basic operations. Yes, we use the swagger from `azure-rest-api-specs` repo but they are too big to use in the template. Later in the doc I mentioned to remove the swagger folder, also I mentioned in the autorest.md section that add swagger to your service and gave an example.